plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol

```diff
@@ -15,19 +15,21 @@ import "../../utils/OnlyFromAddress.sol";
  */
 contract PaymentExitGame is IExitProcessor, OnlyFromAddress, PaymentStandardExitRouter, PaymentInFlightExitRouter {
     PlasmaFramework private plasmaFramework;
+    bool private initDone = false;
 
     /**
      * @dev use struct PaymentExitGameArgs to avoid stack too deep compilation error.
      */
-    constructor(PaymentExitGameArgs.Args memory args)
-        public
-        PaymentStandardExitRouter(args)
-        PaymentInFlightExitRouter(args)
+    function init(PaymentExitGameArgs.Args memory args) public
     {
-        plasmaFramework = args.framework;
-
+        require(!initDone, "Exit game was already initialized");
+        require(msg.sender == args.framework.getMaintainer(), "Only Maintainer can perform this action");
         // makes sure that the spending condition has already renounced ownership
         require(args.spendingConditionRegistry.owner() == address(0), "Spending condition registry ownership needs to be renounced");
+        initDone = true;
+        plasmaFramework = args.framework;
```

Comment:
@thec00n want some input and discussion with you here. We have functions that require to be called via `plasmaFramework`. However, this change make the limitation beneath `onlyFrom(address(plasmaFramework))` not useful before this `init` is called.

However, I _think_ the default value would be zero so without it being called, no one should be able to call via zero address to other functions. Curious on your take here, shall we add the check of `initDone` to those functions? Or it should be fine?